lib/internal/modules/esm/assets/load.js

```diff
@@ -0,0 +1,241 @@
+          if (location.protocol !== 'http:' && location.protocol !== 'https:') {
+            reject(new ERR_UNSUPPORTED_ESM_NETWORK_IMPORT_SPECIFIER(
+              res.headers.location,
+              parsed.href,
+              'cannot redirect to non-network location'));
+            return;
+          }
+          return PromiseResolve(
+            fetchWithRedirects(location)
+          ).then((entry) => {
+            getCache.set(parsed.href, entry);
+            fulfill(entry);
+          });
+
+        }
+      }
+      if (res.statusCode > 303 && res.statusCode < 200) {
+        dispose();
+        reject(
+          new ERR_UNSUPPORTED_ESM_NETWORK_BAD_RESPONSE(
+            'HTTP response returned status code of ' + res.statusCode));
+        return;
+      }
+      const { headers } = res;
+      const contentType = headers['content-type'];
+      if (!contentType) {
+        dispose();
+        reject(new ERR_UNSUPPORTED_ESM_NETWORK_BAD_RESPONSE(
+          'content-type is required'));
+        return;
+      }
+      /**
+       * @type {CacheEntry}
+       */
+      const entry = {
+        resolvedHREF: parsed.href,
+        headers: {
+          'content-type': res.headers['content-type']
+        },
+        body: new Promise((f, r) => {
+          res.on('error', (e) => {
+            dispose();
+            reject(e);
+            r(e);
+          });
+          const buffers = [];
+          let size = 0;
+          res.on('data', (d) => {
+            ArrayPrototypePush(buffers, d);
+            size += d.length;
+          });
+          res.on('end', () => {
+            const body = entry.body = /** @type {Buffer} */(
+              BufferConcat(buffers, size)
+            );
+            f(body);
+          });
+        }),
+      };
+      getCache.set(parsed.href, entry);
+      fulfill(entry);
+    });
+  });
+  getCache.set(parsed.href, result);
+  return result;
+}
+
+/**
+ * Returns if an address has local status by if it is going to a local
+ * interface or is an address resolved by DNS to be a local interface
+ * @param {string} hostname url.hostname to test
+ * @returns {Promise<boolean>}
+ */
+async function isLocalAddress(hostname) {
+  try {
+    if (StringPrototypeStartsWith(hostname, '[')) {
+      hostname = StringPrototypeSlice(hostname, 1, -1);
+    }
+    const addr = await dnsLookup(hostname, { verbatim: true });
+    if (addr.family === 4) {
+      return addr.address === '127.0.0.1';
+    } else if (addr.family === 6) {
+      return normalizeIPv6(addr.address) ===
+        '0000:0000:0000:0000:0000:0000:0000:0001';
+    }
+  } catch {}
+  return false;
+}
+
+/**
+ * Used to expand interface addresses to a canonical form for comparisons
+ * E.G. comparing ::1, 0::1, etc. to 0000:0000:0000:0000:0000:0000:0000:0001
+ * Due to IPv6 having multiple ways to represent a single address we need to
+ * normalize
+ * @param {string} str
+ * @returns {string}
+ */
+function normalizeIPv6(str) {
```

Comment:
I feel like core should expose a function that converts an ip address from its string representation to a bigint or some other binary-based value (out of scope)

Reply:
possible, but why binary instead of string? most APIs taking addresses are expecting strings.

Reply:
I had something like `ipv6ToBigint(address) === 1n` in mind to replace this function. But you're right, maybe we just need an official normalize function that returns a string...